packages/shared/routes/dashboard/settings/views/security/Security.svelte

```diff
@@ -0,0 +1,61 @@
+<script lang="typescript">
+    import { HR } from 'shared/components'
+    import { Electron } from 'shared/lib/electron'
+    import { isSoftwareProfile, updateProfile } from 'shared/lib/profile'
+    import { SecuritySettings } from 'shared/lib/typings/routes'
+    import { getDefaultStrongholdName } from 'shared/lib/utils'
+    import { api } from 'shared/lib/wallet'
+    import { AppLock, ChangePassword, ChangePincode, DeleteProfile, ExportStronghold } from './'
+
+    const settings: {
+        component: unknown
+        childRoute: SecuritySettings
+        requireSoftware?: boolean
+    }[] = [
+        { component: ExportStronghold, childRoute: SecuritySettings.ExportStronghold, requireSoftware: true },
+        { component: AppLock, childRoute: SecuritySettings.AppLock },
+        { component: ChangePassword, childRoute: SecuritySettings.ChangePassword, requireSoftware: true },
+        { component: ChangePincode, childRoute: SecuritySettings.ChangePincode },
+        { component: DeleteProfile, childRoute: SecuritySettings.DeleteProfile },
+    ]
+
+    const props = {
+        [SecuritySettings.ExportStronghold]: { exportStronghold },
+        [SecuritySettings.ChangePassword]: { exportStronghold },
+    }
+
+    function exportStronghold(password: string, callback?: (cancelled: boolean, err?: string) => void) {
+        Electron.getStrongholdBackupDestination(getDefaultStrongholdName())
+            .then((result) => {
+                if (result) {
+                    api.backup(result, password, {
+                        onSuccess() {
+                            updateProfile('lastStrongholdBackupTime', new Date())
+                            callback(false)
+                        },
+                        onError(err) {
+                            callback(false, err.error)
+                        },
+                    })
+                } else {
+                    callback(true)
+                }
+            })
+            .catch((err) => {
+                callback(false, err.error)
+            })
+    }
```

Comment:
Is there a reason we export the back up when the password changes, just to spoon feed the user? If so and we use it in multiple places I would move this function out of the view component and into a library.